// @angular/material/button/button.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from '@angular/material/button/button';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/material/core/ripple/ripple';
import * as import4 from '@angular/common/src/localization';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/i18n/tokens';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from '@angular/core/src/linker/view';
import * as import9 from '@angular/core/src/linker/view_utils';
import * as import10 from '@angular/core/src/linker/debug_context';
import * as import11 from '@angular/core/src/render/api';
import * as import12 from '@angular/core/src/metadata/view';
import * as import13 from '@angular/core/src/linker/view_type';
import * as import14 from '@angular/core/src/linker/component_factory';
import * as import15 from '@angular/core/src/linker/element_ref';
import * as import16 from './button.css';
import * as import17 from '@angular/core/src/linker/template_ref';
import * as import18 from '@angular/common/src/directives/ng_if';
import * as import19 from '@angular/core/src/linker/view_container';
import * as import20 from '../../common/src/directives/ng_if.ngfactory';
import * as import21 from '../core/ripple/ripple.ngfactory';
class MdButtonModuleInjector extends import0.NgModuleInjector<import1.MdButtonModule> {
  _CommonModule_0:import2.CommonModule;
  _MdRippleModule_1:import3.MdRippleModule;
  _MdButtonModule_2:import1.MdButtonModule;
  __NgLocalization_3:import4.NgLocaleLocalization;
  constructor(parent:import5.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import4.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import4.NgLocaleLocalization(this.parent.get(import6.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.MdButtonModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._MdRippleModule_1 = new import3.MdRippleModule();
    this._MdButtonModule_2 = new import1.MdButtonModule();
    return this._MdButtonModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.MdRippleModule)) { return this._MdRippleModule_1; }
    if ((token === import1.MdButtonModule)) { return this._MdButtonModule_2; }
    if ((token === import4.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const MdButtonModuleNgFactory:import0.NgModuleFactory<import1.MdButtonModule> = new import0.NgModuleFactory(MdButtonModuleInjector,import1.MdButtonModule);
export class Wrapper_MdButton {
  /*private*/ _eventHandler:Function;
  context:import1.MdButton;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import1.MdButton(p0,p1);
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
  }
  ngOnDetach(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  check_disableRipple(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.disableRipple = currValue;
      this._expr_0 = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.disabled = currValue;
      this._expr_1 = currValue;
    }
  }
  check_color(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.color = currValue;
      this._expr_2 = currValue;
    }
  }
  ngDoCheck(view:import8.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any,throwOnChange:boolean):void {
    const currVal_3:any = this.context.disabled;
    if (import9.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      view.renderer.setElementProperty(el,'disabled',currVal_3);
      this._expr_3 = currVal_3;
    }
    const currVal_4:any = this.context._isKeyboardFocused;
    if (import9.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      view.renderer.setElementClass(el,'md-button-focus',currVal_4);
      this._expr_4 = currVal_4;
    }
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    if ((eventName == 'mousedown')) {
      const pd_sub_0:any = ((<any>this.context._setMousedown()) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'focus')) {
      const pd_sub_1:any = ((<any>this.context._setKeyboardFocus()) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_2:any = ((<any>this.context._removeKeyboardFocus()) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
  subscribe(view:import8.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
export class Wrapper_MdAnchor {
  /*private*/ _eventHandler:Function;
  context:import1.MdAnchor;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import1.MdAnchor(p0,p1);
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
    this._expr_5 = import7.UNINITIALIZED;
    this._expr_6 = import7.UNINITIALIZED;
  }
  ngOnDetach(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  check_color(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.color = currValue;
      this._expr_0 = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.disabled = currValue;
      this._expr_1 = currValue;
    }
  }
  check_disableRipple(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.disableRipple = currValue;
      this._expr_2 = currValue;
    }
  }
  ngDoCheck(view:import8.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any,throwOnChange:boolean):void {
    const currVal_3:any = this.context.disabled;
    if (import9.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      view.renderer.setElementAttribute(el,'disabled',((currVal_3 == null)? (null as any): currVal_3.toString()));
      this._expr_3 = currVal_3;
    }
    const currVal_4:any = this.context._isKeyboardFocused;
    if (import9.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      view.renderer.setElementClass(el,'md-button-focus',currVal_4);
      this._expr_4 = currVal_4;
    }
    const currVal_5:any = this.context.tabIndex;
    if (import9.checkBinding(throwOnChange,this._expr_5,currVal_5)) {
      view.renderer.setElementProperty(el,'tabIndex',currVal_5);
      this._expr_5 = currVal_5;
    }
    const currVal_6:any = this.context.isAriaDisabled;
    if (import9.checkBinding(throwOnChange,this._expr_6,currVal_6)) {
      view.renderer.setElementAttribute(el,'aria-disabled',((currVal_6 == null)? (null as any): currVal_6.toString()));
      this._expr_6 = currVal_6;
    }
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    if ((eventName == 'mousedown')) {
      const pd_sub_0:any = ((<any>this.context._setMousedown()) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'focus')) {
      const pd_sub_1:any = ((<any>this.context._setKeyboardFocus()) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_2:any = ((<any>this.context._removeKeyboardFocus()) !== false);
      result = (pd_sub_2 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_3:any = ((<any>this.context._haltDisabledEvents($event)) !== false);
      result = (pd_sub_3 && result);
    }
    return result;
  }
  subscribe(view:import8.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
const nodeDebugInfos_MdButton_Host0:import10.StaticNodeDebugInfo[] = [new import10.StaticNodeDebugInfo([import1.MdButton],import1.MdButton,{})];
var renderType_MdButton_Host:import11.RenderComponentType = import9.createRenderComponentType('',0,import12.ViewEncapsulation.None,([] as any[]),{});
class View_MdButton_Host0 extends import8.DebugAppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.MdButton>;
  _MdButton_0_3:Wrapper_MdButton;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_MdButton_Host0,renderType_MdButton_Host,import13.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_MdButton_Host0);
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import9.selectOrCreateRenderHostElement(this.renderer,'button',new import9.InlineArray2(2,'md-button',''),rootSelector,this.debug(0,0,0));
    this.compView_0 = new View_MdButton0(this.viewUtils,this,0,this._el_0);
    this._MdButton_0_3 = new Wrapper_MdButton(new import15.ElementRef(this._el_0),this.renderer);
    this.compView_0.create(this._MdButton_0_3.context);
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray8(6,'mousedown',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return new import14.ComponentRef_<any>(0,this,this._el_0,this._MdButton_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.MdButton) && (0 === requestNodeIndex))) { return this._MdButton_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,0,0);
    if (this._MdButton_0_3.ngDoCheck(this,this._el_0,throwOnChange)) { this.compView_0.markAsCheckOnce(); }
    this._MdButton_0_3.checkHost(this,this.compView_0,this._el_0,throwOnChange);
    this.compView_0.detectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.debug(0,0,0);
    this.compView_0.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._MdButton_0_3.handleEvent(eventName,$event) && result);
    return result;
  }
}
export const MdButtonNgFactory:import14.ComponentFactory<import1.MdButton> = new import14.ComponentFactory<import1.MdButton>('button[md-button], button[md-raised-button], button[md-icon-button], button[md-fab], button[md-mini-fab]',View_MdButton_Host0,import1.MdButton);
const styles_MdButton:any[] = [import16.styles];
const nodeDebugInfos_MdButton0:import10.StaticNodeDebugInfo[] = [
  new import10.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import10.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import10.StaticNodeDebugInfo([
    import17.TemplateRef,
    import18.NgIf
  ]
  ,(null as any),{}),
  new import10.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
var renderType_MdButton:import11.RenderComponentType = import9.createRenderComponentType('/Users/jessebowden/ichuwt/frontend/node_modules/@angular/material/button/button.html',1,import12.ViewEncapsulation.None,styles_MdButton,{});
export class View_MdButton0 extends import8.DebugAppView<import1.MdButton> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import19.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import20.Wrapper_NgIf;
  _text_3:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_MdButton0,renderType_MdButton,import13.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckOnce,nodeDebugInfos_MdButton0);
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._el_0 = import9.createRenderElement(this.renderer,parentRenderNode,'span',new import9.InlineArray2(2,'class','md-button-wrapper'),this.debug(0,0,0));
    this.debug((null as any),0,32);
    this.projectNodes(this._el_0,0);
    this._text_1 = this.renderer.createText(parentRenderNode,'\n',this.debug(1,0,64));
    this._anchor_2 = this.renderer.createTemplateAnchor(parentRenderNode,this.debug(2,1,0));
    this._vc_2 = new import19.ViewContainer(2,(null as any),this,this._anchor_2);
    this._TemplateRef_2_5 = new import17.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import20.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(parentRenderNode,'\n',this.debug(3,5,56));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import17.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import18.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(2,1,15);
    const currVal_2_0_0:boolean = !this.context._isRippleDisabled();
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 2)) { return new View_MdButton1(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
const nodeDebugInfos_MdButton1:import10.StaticNodeDebugInfo[] = [new import10.StaticNodeDebugInfo([import3.MdRipple],(null as any),{})];
class View_MdButton1 extends import8.DebugAppView<any> {
  _el_0:any;
  _MdRipple_0_3:import21.Wrapper_MdRipple;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import19.ViewContainer) {
    super(View_MdButton1,renderType_MdButton,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_MdButton1,declaredViewContainer);
    this._expr_2 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'div',new import9.InlineArray8(6,'class','md-button-ripple','md-ripple','','md-ripple-background-color','rgba(0, 0, 0, 0)'),this.debug(0,1,0));
    this._MdRipple_0_3 = new import21.Wrapper_MdRipple(new import15.ElementRef(this._el_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import3.MdRipple) && (0 === requestNodeIndex))) { return this._MdRipple_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,3,4);
    const currVal_0_0_0:any = this.parentView.context.getHostElement();
    this._MdRipple_0_3.check_trigger(currVal_0_0_0,throwOnChange,false);
    this.debug(0,4,4);
    const currVal_0_0_1:any = (this.parentView.context.isRoundButton()? 'rgba(255, 255, 255, 0.2)': '');
    this._MdRipple_0_3.check_color(currVal_0_0_1,throwOnChange,false);
    this.debug(0,5,4);
    const currVal_0_0_2:any = 'rgba(0, 0, 0, 0)';
    this._MdRipple_0_3.check_backgroundColor(currVal_0_0_2,throwOnChange,false);
    this._MdRipple_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.debug(0,2,4);
    const currVal_2:any = this.parentView.context.isRoundButton();
    if (import9.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementClass(this._el_0,'md-button-ripple-round',currVal_2);
      this._expr_2 = currVal_2;
    }
    this._MdRipple_0_3.checkHost(this,this,this._el_0,throwOnChange);
  }
  destroyInternal():void {
    this._MdRipple_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
const nodeDebugInfos_MdAnchor_Host0:import10.StaticNodeDebugInfo[] = [new import10.StaticNodeDebugInfo([import1.MdAnchor],import1.MdAnchor,{})];
var renderType_MdAnchor_Host:import11.RenderComponentType = import9.createRenderComponentType('',0,import12.ViewEncapsulation.None,([] as any[]),{});
class View_MdAnchor_Host0 extends import8.DebugAppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.MdAnchor>;
  _MdAnchor_0_3:Wrapper_MdAnchor;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_MdAnchor_Host0,renderType_MdAnchor_Host,import13.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_MdAnchor_Host0);
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import9.selectOrCreateRenderHostElement(this.renderer,'a',new import9.InlineArray2(2,'md-button',''),rootSelector,this.debug(0,0,0));
    this.compView_0 = new View_MdAnchor0(this.viewUtils,this,0,this._el_0);
    this._MdAnchor_0_3 = new Wrapper_MdAnchor(new import15.ElementRef(this._el_0),this.renderer);
    this.compView_0.create(this._MdAnchor_0_3.context);
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray8(8,'mousedown',(null as any),'focus',(null as any),'blur',(null as any),'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return new import14.ComponentRef_<any>(0,this,this._el_0,this._MdAnchor_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.MdAnchor) && (0 === requestNodeIndex))) { return this._MdAnchor_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,0,0);
    this._MdAnchor_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this._MdAnchor_0_3.checkHost(this,this.compView_0,this._el_0,throwOnChange);
    this.compView_0.detectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.debug(0,0,0);
    this.compView_0.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._MdAnchor_0_3.handleEvent(eventName,$event) && result);
    return result;
  }
}
export const MdAnchorNgFactory:import14.ComponentFactory<import1.MdAnchor> = new import14.ComponentFactory<import1.MdAnchor>('a[md-button], a[md-raised-button], a[md-icon-button], a[md-fab], a[md-mini-fab]',View_MdAnchor_Host0,import1.MdAnchor);
const styles_MdAnchor:any[] = [import16.styles];
const nodeDebugInfos_MdAnchor0:import10.StaticNodeDebugInfo[] = [
  new import10.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import10.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import10.StaticNodeDebugInfo([
    import17.TemplateRef,
    import18.NgIf
  ]
  ,(null as any),{}),
  new import10.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
var renderType_MdAnchor:import11.RenderComponentType = import9.createRenderComponentType('/Users/jessebowden/ichuwt/frontend/node_modules/@angular/material/button/button.html',1,import12.ViewEncapsulation.None,styles_MdAnchor,{});
export class View_MdAnchor0 extends import8.DebugAppView<import1.MdAnchor> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import19.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import20.Wrapper_NgIf;
  _text_3:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_MdAnchor0,renderType_MdAnchor,import13.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_MdAnchor0);
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._el_0 = import9.createRenderElement(this.renderer,parentRenderNode,'span',new import9.InlineArray2(2,'class','md-button-wrapper'),this.debug(0,0,0));
    this.debug((null as any),0,32);
    this.projectNodes(this._el_0,0);
    this._text_1 = this.renderer.createText(parentRenderNode,'\n',this.debug(1,0,64));
    this._anchor_2 = this.renderer.createTemplateAnchor(parentRenderNode,this.debug(2,1,0));
    this._vc_2 = new import19.ViewContainer(2,(null as any),this,this._anchor_2);
    this._TemplateRef_2_5 = new import17.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import20.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(parentRenderNode,'\n',this.debug(3,5,56));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import17.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import18.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(2,1,15);
    const currVal_2_0_0:boolean = !this.context._isRippleDisabled();
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 2)) { return new View_MdAnchor1(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
const nodeDebugInfos_MdAnchor1:import10.StaticNodeDebugInfo[] = [new import10.StaticNodeDebugInfo([import3.MdRipple],(null as any),{})];
class View_MdAnchor1 extends import8.DebugAppView<any> {
  _el_0:any;
  _MdRipple_0_3:import21.Wrapper_MdRipple;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import19.ViewContainer) {
    super(View_MdAnchor1,renderType_MdAnchor,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_MdAnchor1,declaredViewContainer);
    this._expr_2 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'div',new import9.InlineArray8(6,'class','md-button-ripple','md-ripple','','md-ripple-background-color','rgba(0, 0, 0, 0)'),this.debug(0,1,0));
    this._MdRipple_0_3 = new import21.Wrapper_MdRipple(new import15.ElementRef(this._el_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import3.MdRipple) && (0 === requestNodeIndex))) { return this._MdRipple_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,3,4);
    const currVal_0_0_0:any = this.parentView.context.getHostElement();
    this._MdRipple_0_3.check_trigger(currVal_0_0_0,throwOnChange,false);
    this.debug(0,4,4);
    const currVal_0_0_1:any = (this.parentView.context.isRoundButton()? 'rgba(255, 255, 255, 0.2)': '');
    this._MdRipple_0_3.check_color(currVal_0_0_1,throwOnChange,false);
    this.debug(0,5,4);
    const currVal_0_0_2:any = 'rgba(0, 0, 0, 0)';
    this._MdRipple_0_3.check_backgroundColor(currVal_0_0_2,throwOnChange,false);
    this._MdRipple_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.debug(0,2,4);
    const currVal_2:any = this.parentView.context.isRoundButton();
    if (import9.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementClass(this._el_0,'md-button-ripple-round',currVal_2);
      this._expr_2 = currVal_2;
    }
    this._MdRipple_0_3.checkHost(this,this,this._el_0,throwOnChange);
  }
  destroyInternal():void {
    this._MdRipple_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
