// @angular/material/core/a11y/focus-trap.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/material/core/a11y/focus-trap';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/debug_context';
import * as import3 from '@angular/core/src/render/api';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from '@angular/core/src/linker/component_factory';
import * as import9 from '@angular/material/core/a11y/interactivity-checker';
import * as import10 from '@angular/core/src/linker/query_list';
import * as import11 from '@angular/core/src/linker/element_ref';
export class Wrapper_FocusTrap {
  /*private*/ _eventHandler:Function;
  context:import0.FocusTrap;
  /*private*/ _changed:boolean;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import0.FocusTrap(p0);
  }
  ngOnDetach(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  ngDoCheck(view:import1.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import1.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
const nodeDebugInfos_FocusTrap_Host0:import2.StaticNodeDebugInfo[] = [new import2.StaticNodeDebugInfo([import0.FocusTrap],import0.FocusTrap,{})];
var renderType_FocusTrap_Host:import3.RenderComponentType = import4.createRenderComponentType('',0,import5.ViewEncapsulation.None,([] as any[]),{});
class View_FocusTrap_Host0 extends import1.DebugAppView<any> {
  _el_0:any;
  compView_0:import1.AppView<import0.FocusTrap>;
  _FocusTrap_0_3:Wrapper_FocusTrap;
  constructor(viewUtils:import4.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_FocusTrap_Host0,renderType_FocusTrap_Host,import6.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_FocusTrap_Host0);
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import4.selectOrCreateRenderHostElement(this.renderer,'focus-trap',import4.EMPTY_INLINE_ARRAY,rootSelector,this.debug(0,0,0));
    this.compView_0 = new View_FocusTrap0(this.viewUtils,this,0,this._el_0);
    this._FocusTrap_0_3 = new Wrapper_FocusTrap(this.injectorGet(import9.InteractivityChecker,this.parentIndex));
    this.compView_0.create(this._FocusTrap_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import8.ComponentRef_<any>(0,this,this._el_0,this._FocusTrap_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.FocusTrap) && (0 === requestNodeIndex))) { return this._FocusTrap_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,0,0);
    this._FocusTrap_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.detectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
  }
}
export const FocusTrapNgFactory:import8.ComponentFactory<import0.FocusTrap> = new import8.ComponentFactory<import0.FocusTrap>('focus-trap',View_FocusTrap_Host0,import0.FocusTrap);
const styles_FocusTrap:any[] = ([] as any[]);
const nodeDebugInfos_FocusTrap0:import2.StaticNodeDebugInfo[] = [
  new import2.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import2.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import2.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import2.StaticNodeDebugInfo(([] as any[]),(null as any),{trappedContent: (null as any)}),
  new import2.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import2.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
var renderType_FocusTrap:import3.RenderComponentType = import4.createRenderComponentType('/Users/jessebowden/ichuwt/frontend/node_modules/@angular/material/core/a11y/focus-trap.d.ts class FocusTrap - inline template',1,import5.ViewEncapsulation.None,styles_FocusTrap,{});
export class View_FocusTrap0 extends import1.DebugAppView<import0.FocusTrap> {
  _viewQuery_trappedContent_0:import10.QueryList<any>;
  _text_0:any;
  _el_1:any;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  constructor(viewUtils:import4.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_FocusTrap0,renderType_FocusTrap,import6.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_FocusTrap0);
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._viewQuery_trappedContent_0 = new import10.QueryList<any>();
    this._text_0 = this.renderer.createText(parentRenderNode,'\n  ',this.debug(0,0,0));
    this._el_1 = import4.createRenderElement(this.renderer,parentRenderNode,'div',new import4.InlineArray2(2,'tabindex','0'),this.debug(1,1,2));
    this._text_2 = this.renderer.createText(parentRenderNode,'\n  ',this.debug(2,1,63));
    this._el_3 = import4.createRenderElement(this.renderer,parentRenderNode,'div',import4.EMPTY_INLINE_ARRAY,this.debug(3,2,2));
    this.debug((null as any),2,23);
    this.projectNodes(this._el_3,0);
    this._text_4 = this.renderer.createText(parentRenderNode,'\n  ',this.debug(4,2,54));
    this._el_5 = import4.createRenderElement(this.renderer,parentRenderNode,'div',new import4.InlineArray2(2,'tabindex','0'),this.debug(5,3,2));
    var disposable_0:Function = import4.subscribeToRenderElement(this,this._el_1,new import4.InlineArray2(2,'focus',(null as any)),this.eventHandler(this.handleEvent_1));
    var disposable_1:Function = import4.subscribeToRenderElement(this,this._el_5,new import4.InlineArray2(2,'focus',(null as any)),this.eventHandler(this.handleEvent_5));
    this._viewQuery_trappedContent_0.reset([new import11.ElementRef(this._el_3)]);
    this.context.trappedContent = this._viewQuery_trappedContent_0.first;
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5
    ]
    ),[
      disposable_0,
      disposable_1
    ]
    );
    return (null as any);
  }
  handleEvent_1(eventName:string,$event:any):boolean {
    this.debug(1,1,2);
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'focus')) {
      const pd_sub_0:any = ((<any>this.context.focusLastTabbableElement()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_5(eventName:string,$event:any):boolean {
    this.debug(5,3,2);
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'focus')) {
      const pd_sub_0:any = ((<any>this.context.focusFirstTabbableElement()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
