// @angular/material/snack-bar/simple-snack-bar.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/material/snack-bar/simple-snack-bar';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/debug_context';
import * as import3 from '@angular/core/src/render/api';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from '@angular/core/src/linker/component_factory';
import * as import9 from './simple-snack-bar.css.shim';
import * as import10 from '@angular/core/src/linker/template_ref';
import * as import11 from '@angular/common/src/directives/ng_if';
import * as import12 from '@angular/core/src/linker/view_container';
import * as import13 from '../../common/src/directives/ng_if.ngfactory';
export class Wrapper_SimpleSnackBar {
  /*private*/ _eventHandler:Function;
  context:import0.SimpleSnackBar;
  /*private*/ _changed:boolean;
  constructor() {
    this._changed = false;
    this.context = new import0.SimpleSnackBar();
  }
  ngOnDetach(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  ngDoCheck(view:import1.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import1.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
const nodeDebugInfos_SimpleSnackBar_Host0:import2.StaticNodeDebugInfo[] = [new import2.StaticNodeDebugInfo([import0.SimpleSnackBar],import0.SimpleSnackBar,{})];
var renderType_SimpleSnackBar_Host:import3.RenderComponentType = import4.createRenderComponentType('',0,import5.ViewEncapsulation.None,([] as any[]),{});
class View_SimpleSnackBar_Host0 extends import1.DebugAppView<any> {
  _el_0:any;
  compView_0:import1.AppView<import0.SimpleSnackBar>;
  _SimpleSnackBar_0_3:Wrapper_SimpleSnackBar;
  constructor(viewUtils:import4.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_SimpleSnackBar_Host0,renderType_SimpleSnackBar_Host,import6.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_SimpleSnackBar_Host0);
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import4.selectOrCreateRenderHostElement(this.renderer,'simple-snack-bar',import4.EMPTY_INLINE_ARRAY,rootSelector,this.debug(0,0,0));
    this.compView_0 = new View_SimpleSnackBar0(this.viewUtils,this,0,this._el_0);
    this._SimpleSnackBar_0_3 = new Wrapper_SimpleSnackBar();
    this.compView_0.create(this._SimpleSnackBar_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import8.ComponentRef_<any>(0,this,this._el_0,this._SimpleSnackBar_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.SimpleSnackBar) && (0 === requestNodeIndex))) { return this._SimpleSnackBar_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,0,0);
    this._SimpleSnackBar_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.detectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const SimpleSnackBarNgFactory:import8.ComponentFactory<import0.SimpleSnackBar> = new import8.ComponentFactory<import0.SimpleSnackBar>('simple-snack-bar',View_SimpleSnackBar_Host0,import0.SimpleSnackBar);
const styles_SimpleSnackBar:any[] = [import9.styles];
const nodeDebugInfos_SimpleSnackBar0:import2.StaticNodeDebugInfo[] = [
  new import2.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import2.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import2.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import2.StaticNodeDebugInfo([
    import10.TemplateRef,
    import11.NgIf
  ]
  ,(null as any),{})
]
;
var renderType_SimpleSnackBar:import3.RenderComponentType = import4.createRenderComponentType('/Users/jessebowden/ichuwt/frontend/node_modules/@angular/material/snack-bar/simple-snack-bar.html',0,import5.ViewEncapsulation.Emulated,styles_SimpleSnackBar,{});
export class View_SimpleSnackBar0 extends import1.DebugAppView<import0.SimpleSnackBar> {
  _el_0:any;
  _text_1:any;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import12.ViewContainer;
  _TemplateRef_3_5:any;
  _NgIf_3_6:import13.Wrapper_NgIf;
  /*private*/ _expr_7:any;
  constructor(viewUtils:import4.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_SimpleSnackBar0,renderType_SimpleSnackBar,import6.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_SimpleSnackBar0);
    this._expr_7 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._el_0 = import4.createRenderElement(this.renderer,parentRenderNode,'span',new import4.InlineArray2(2,'class','md-simple-snackbar-message'),this.debug(0,0,0));
    this._text_1 = this.renderer.createText(this._el_0,'',this.debug(1,0,41));
    this._text_2 = this.renderer.createText(parentRenderNode,'\n',this.debug(2,0,59));
    this._anchor_3 = this.renderer.createTemplateAnchor(parentRenderNode,this.debug(3,1,0));
    this._vc_3 = new import12.ViewContainer(3,(null as any),this,this._anchor_3);
    this._TemplateRef_3_5 = new import10.TemplateRef_(this,3,this._anchor_3);
    this._NgIf_3_6 = new import13.Wrapper_NgIf(this._vc_3.vcRef,this._TemplateRef_3_5);
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._text_2,
      this._anchor_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import10.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import11.NgIf) && (3 === requestNodeIndex))) { return this._NgIf_3_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(3,2,8);
    const currVal_3_0_0:any = this.context.hasAction;
    this._NgIf_3_6.check_ngIf(currVal_3_0_0,throwOnChange,false);
    this._NgIf_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
    this.debug(1,0,41);
    const currVal_7:any = import4.inlineInterpolate(1,'',this.context.message,'');
    if (import4.checkBinding(throwOnChange,this._expr_7,currVal_7)) {
      this.renderer.setText(this._text_1,currVal_7);
      this._expr_7 = currVal_7;
    }
  }
  destroyInternal():void {
    this._vc_3.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import1.AppView<any> {
    if ((nodeIndex == 3)) { return new View_SimpleSnackBar1(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    return (null as any);
  }
}
const nodeDebugInfos_SimpleSnackBar1:import2.StaticNodeDebugInfo[] = [
  new import2.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import2.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
class View_SimpleSnackBar1 extends import1.DebugAppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import4.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import12.ViewContainer) {
    super(View_SimpleSnackBar1,renderType_SimpleSnackBar,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_SimpleSnackBar1,declaredViewContainer);
    this._expr_2 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'button',new import4.InlineArray4(4,'class','md-simple-snackbar-action','md-button',''),this.debug(0,1,0));
    this._text_1 = this.renderer.createText(this._el_0,'',this.debug(1,2,46));
    var disposable_0:Function = import4.subscribeToRenderElement(this,this._el_0,new import4.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),[disposable_0]);
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(1,2,46);
    const currVal_2:any = import4.inlineInterpolate(1,'',this.parentView.context.action,'');
    if (import4.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.debug(0,1,0);
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.dismiss()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
