// @angular/material/core/overlay/overlay-directives.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from '@angular/material/core/overlay/overlay-directives';
import * as import2 from '@angular/material/core/portal/portal-directives';
import * as import3 from '@angular/core/src/di/injector';
import * as import4 from '@angular/core/src/change_detection/change_detection';
import * as import5 from '@angular/core/src/linker/view';
import * as import6 from '@angular/core/src/linker/view_utils';
class OverlayModuleInjector extends import0.NgModuleInjector<import1.OverlayModule> {
  _PortalModule_0:import2.PortalModule;
  _OverlayModule_1:import1.OverlayModule;
  constructor(parent:import3.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  createInternal():import1.OverlayModule {
    this._PortalModule_0 = new import2.PortalModule();
    this._OverlayModule_1 = new import1.OverlayModule();
    return this._OverlayModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.PortalModule)) { return this._PortalModule_0; }
    if ((token === import1.OverlayModule)) { return this._OverlayModule_1; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const OverlayModuleNgFactory:import0.NgModuleFactory<import1.OverlayModule> = new import0.NgModuleFactory(OverlayModuleInjector,import1.OverlayModule);
export class Wrapper_ConnectedOverlayDirective {
  /*private*/ _eventHandler:Function;
  context:import1.ConnectedOverlayDirective;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  subscription0:any;
  constructor(p0:any,p1:any,p2:any,p3:any) {
    this._changed = false;
    this.context = new import1.ConnectedOverlayDirective(p0,p1,p2,p3);
    this._expr_0 = import4.UNINITIALIZED;
    this._expr_1 = import4.UNINITIALIZED;
    this._expr_2 = import4.UNINITIALIZED;
    this._expr_3 = import4.UNINITIALIZED;
    this._expr_4 = import4.UNINITIALIZED;
    this._expr_5 = import4.UNINITIALIZED;
    this._expr_6 = import4.UNINITIALIZED;
    this._expr_7 = import4.UNINITIALIZED;
    this._expr_8 = import4.UNINITIALIZED;
  }
  ngOnDetach(view:import5.AppView<any>,componentView:import5.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
    (this.subscription0 && this.subscription0.unsubscribe());
  }
  check_origin(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import6.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.origin = currValue;
      this._expr_0 = currValue;
    }
  }
  check_positions(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import6.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.positions = currValue;
      this._expr_1 = currValue;
    }
  }
  check_offsetX(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import6.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.offsetX = currValue;
      this._expr_2 = currValue;
    }
  }
  check_offsetY(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import6.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.offsetY = currValue;
      this._expr_3 = currValue;
    }
  }
  check_width(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import6.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.width = currValue;
      this._expr_4 = currValue;
    }
  }
  check_height(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import6.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.height = currValue;
      this._expr_5 = currValue;
    }
  }
  check_backdropClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import6.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.backdropClass = currValue;
      this._expr_6 = currValue;
    }
  }
  check_hasBackdrop(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import6.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.hasBackdrop = currValue;
      this._expr_7 = currValue;
    }
  }
  check_open(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import6.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.open = currValue;
      this._expr_8 = currValue;
    }
  }
  ngDoCheck(view:import5.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import5.AppView<any>,componentView:import5.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import5.AppView<any>,_eventHandler:any,emit0:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.backdropClick.subscribe(_eventHandler.bind(view,'backdropClick'))); }
  }
}
export class Wrapper_OverlayOrigin {
  /*private*/ _eventHandler:Function;
  context:import1.OverlayOrigin;
  /*private*/ _changed:boolean;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import1.OverlayOrigin(p0);
  }
  ngOnDetach(view:import5.AppView<any>,componentView:import5.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  ngDoCheck(view:import5.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import5.AppView<any>,componentView:import5.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import5.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
