// @angular/material/core/line/line.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from '@angular/material/core/line/line';
import * as import2 from '@angular/core/src/di/injector';
import * as import3 from '@angular/core/src/linker/view';
class MdLineModuleInjector extends import0.NgModuleInjector<import1.MdLineModule> {
  _MdLineModule_0:import1.MdLineModule;
  constructor(parent:import2.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  createInternal():import1.MdLineModule {
    this._MdLineModule_0 = new import1.MdLineModule();
    return this._MdLineModule_0;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import1.MdLineModule)) { return this._MdLineModule_0; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const MdLineModuleNgFactory:import0.NgModuleFactory<import1.MdLineModule> = new import0.NgModuleFactory(MdLineModuleInjector,import1.MdLineModule);
export class Wrapper_MdLine {
  /*private*/ _eventHandler:Function;
  context:import1.MdLine;
  /*private*/ _changed:boolean;
  constructor() {
    this._changed = false;
    this.context = new import1.MdLine();
  }
  ngOnDetach(view:import3.AppView<any>,componentView:import3.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  ngDoCheck(view:import3.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import3.AppView<any>,componentView:import3.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import3.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
