// @angular/material/grid-list/grid-tile.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/material/grid-list/grid-tile';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/linker/debug_context';
import * as import5 from '@angular/core/src/render/api';
import * as import6 from '@angular/core/src/metadata/view';
import * as import7 from '@angular/core/src/linker/view_type';
import * as import8 from '@angular/core/src/linker/component_factory';
import * as import9 from '@angular/core/src/linker/element_ref';
import * as import10 from './grid-list.css';
import * as import11 from '@angular/core/src/linker/query_list';
export class Wrapper_MdGridTile {
  /*private*/ _eventHandler:Function;
  context:import0.MdGridTile;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import0.MdGridTile(p0,p1);
    this._expr_0 = import1.UNINITIALIZED;
    this._expr_1 = import1.UNINITIALIZED;
  }
  ngOnDetach(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  check_rowspan(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.rowspan = currValue;
      this._expr_0 = currValue;
    }
  }
  check_colspan(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.colspan = currValue;
      this._expr_1 = currValue;
    }
  }
  ngDoCheck(view:import2.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import2.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
export class Wrapper_MdGridTileText {
  /*private*/ _eventHandler:Function;
  context:import0.MdGridTileText;
  /*private*/ _changed:boolean;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import0.MdGridTileText(p0,p1);
  }
  ngOnDetach(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  ngDoCheck(view:import2.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import2.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
const nodeDebugInfos_MdGridTile_Host0:import4.StaticNodeDebugInfo[] = [new import4.StaticNodeDebugInfo([import0.MdGridTile],import0.MdGridTile,{})];
var renderType_MdGridTile_Host:import5.RenderComponentType = import3.createRenderComponentType('',0,import6.ViewEncapsulation.None,([] as any[]),{});
class View_MdGridTile_Host0 extends import2.DebugAppView<any> {
  _el_0:any;
  compView_0:import2.AppView<import0.MdGridTile>;
  _MdGridTile_0_3:Wrapper_MdGridTile;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_MdGridTile_Host0,renderType_MdGridTile_Host,import7.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_MdGridTile_Host0);
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer,'md-grid-tile',new import3.InlineArray2(2,'role','listitem'),rootSelector,this.debug(0,0,0));
    this.compView_0 = new View_MdGridTile0(this.viewUtils,this,0,this._el_0);
    this._MdGridTile_0_3 = new Wrapper_MdGridTile(this.renderer,new import9.ElementRef(this._el_0));
    this.compView_0.create(this._MdGridTile_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import8.ComponentRef_<any>(0,this,this._el_0,this._MdGridTile_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.MdGridTile) && (0 === requestNodeIndex))) { return this._MdGridTile_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,0,0);
    this._MdGridTile_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.detectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
  }
}
export const MdGridTileNgFactory:import8.ComponentFactory<import0.MdGridTile> = new import8.ComponentFactory<import0.MdGridTile>('md-grid-tile',View_MdGridTile_Host0,import0.MdGridTile);
const styles_MdGridTile:any[] = [import10.styles];
const nodeDebugInfos_MdGridTile0:import4.StaticNodeDebugInfo[] = [
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
var renderType_MdGridTile:import5.RenderComponentType = import3.createRenderComponentType('/Users/jessebowden/ichuwt/frontend/node_modules/@angular/material/grid-list/grid-tile.html',1,import6.ViewEncapsulation.None,styles_MdGridTile,{});
export class View_MdGridTile0 extends import2.DebugAppView<import0.MdGridTile> {
  _text_0:any;
  _el_1:any;
  _text_2:any;
  _text_3:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_MdGridTile0,renderType_MdGridTile,import7.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_MdGridTile0);
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n',this.debug(0,0,49));
    this._el_1 = import3.createRenderElement(this.renderer,parentRenderNode,'figure',import3.EMPTY_INLINE_ARRAY,this.debug(1,1,0));
    this._text_2 = this.renderer.createText(this._el_1,'\n  ',this.debug(2,1,8));
    this.debug((null as any),2,2);
    this.projectNodes(this._el_1,0);
    this._text_3 = this.renderer.createText(this._el_1,'\n',this.debug(3,2,27));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
}
const nodeDebugInfos_MdGridTileText_Host0:import4.StaticNodeDebugInfo[] = [new import4.StaticNodeDebugInfo([import0.MdGridTileText],import0.MdGridTileText,{})];
var renderType_MdGridTileText_Host:import5.RenderComponentType = import3.createRenderComponentType('',0,import6.ViewEncapsulation.None,([] as any[]),{});
class View_MdGridTileText_Host0 extends import2.DebugAppView<any> {
  _el_0:any;
  compView_0:import2.AppView<import0.MdGridTileText>;
  _MdGridTileText_0_3:Wrapper_MdGridTileText;
  _query_MdLine_0_0:import11.QueryList<any>;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_MdGridTileText_Host0,renderType_MdGridTileText_Host,import7.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_MdGridTileText_Host0);
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer,'md-grid-tile-header',import3.EMPTY_INLINE_ARRAY,rootSelector,this.debug(0,0,0));
    this.compView_0 = new View_MdGridTileText0(this.viewUtils,this,0,this._el_0);
    this._MdGridTileText_0_3 = new Wrapper_MdGridTileText(this.renderer,new import9.ElementRef(this._el_0));
    this._query_MdLine_0_0 = new import11.QueryList<any>();
    this.compView_0.create(this._MdGridTileText_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import8.ComponentRef_<any>(0,this,this._el_0,this._MdGridTileText_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.MdGridTileText) && (0 === requestNodeIndex))) { return this._MdGridTileText_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.debug(0,0,0);
    this._MdGridTileText_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    if (!throwOnChange) {
      if (this._query_MdLine_0_0.dirty) {
        this._query_MdLine_0_0.reset(([] as any[]));
        this._MdGridTileText_0_3.context._lines = this._query_MdLine_0_0;
        this._query_MdLine_0_0.notifyOnChanges();
      }
      this.debug(0,0,0);
      if ((this.numberOfChecks === 0)) { this._MdGridTileText_0_3.context.ngAfterContentInit(); }
    }
    this.compView_0.detectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
    if (((nodeIndex == 0) && (ngContentIndex == 1))) {  }
    if (((nodeIndex == 0) && (ngContentIndex == 2))) {  }
  }
}
export const MdGridTileTextNgFactory:import8.ComponentFactory<import0.MdGridTileText> = new import8.ComponentFactory<import0.MdGridTileText>('md-grid-tile-header, md-grid-tile-footer',View_MdGridTileText_Host0,import0.MdGridTileText);
const styles_MdGridTileText:any[] = ([] as any[]);
const nodeDebugInfos_MdGridTileText0:import4.StaticNodeDebugInfo[] = [
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{}),
  new import4.StaticNodeDebugInfo(([] as any[]),(null as any),{})
]
;
var renderType_MdGridTileText:import5.RenderComponentType = import3.createRenderComponentType('/Users/jessebowden/ichuwt/frontend/node_modules/@angular/material/grid-list/grid-tile-text.html',3,import6.ViewEncapsulation.None,styles_MdGridTileText,{});
export class View_MdGridTileText0 extends import2.DebugAppView<import0.MdGridTileText> {
  _text_0:any;
  _el_1:any;
  _text_2:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_MdGridTileText0,renderType_MdGridTileText,import7.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,nodeDebugInfos_MdGridTileText0);
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this.debug((null as any),0,0);
    this.projectNodes(parentRenderNode,0);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n',this.debug(0,0,51));
    this._el_1 = import3.createRenderElement(this.renderer,parentRenderNode,'div',new import3.InlineArray2(2,'class','md-grid-list-text'),this.debug(1,1,0));
    this.debug((null as any),1,31);
    this.projectNodes(this._el_1,1);
    this._text_2 = this.renderer.createText(parentRenderNode,'\n',this.debug(2,1,81));
    this.debug((null as any),2,0);
    this.projectNodes(parentRenderNode,2);
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2
    ]
    ),(null as any));
    return (null as any);
  }
}
