// @angular/material/core/compatibility/style-compatibility.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from '@angular/material/core/compatibility/style-compatibility';
import * as import2 from '@angular/core/src/di/injector';
import * as import3 from '@angular/core/src/linker/view';
class StyleCompatibilityModuleInjector extends import0.NgModuleInjector<import1.StyleCompatibilityModule> {
  _StyleCompatibilityModule_0:import1.StyleCompatibilityModule;
  constructor(parent:import2.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  createInternal():import1.StyleCompatibilityModule {
    this._StyleCompatibilityModule_0 = new import1.StyleCompatibilityModule();
    return this._StyleCompatibilityModule_0;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import1.StyleCompatibilityModule)) { return this._StyleCompatibilityModule_0; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const StyleCompatibilityModuleNgFactory:import0.NgModuleFactory<import1.StyleCompatibilityModule> = new import0.NgModuleFactory(StyleCompatibilityModuleInjector,import1.StyleCompatibilityModule);
export class Wrapper_StyleCompatibility {
  /*private*/ _eventHandler:Function;
  context:import1.StyleCompatibility;
  /*private*/ _changed:boolean;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import1.StyleCompatibility(p0,p1);
  }
  ngOnDetach(view:import3.AppView<any>,componentView:import3.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  ngDoCheck(view:import3.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import3.AppView<any>,componentView:import3.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import3.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
